edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/commands/KubernetesCrdCommand.cs

```diff
@@ -0,0 +1,263 @@
+                    var combinedConfig = this.combinedConfigProvider.GetCombinedConfig(m, runtime);
+                    CombinedDockerConfig dockerConfig = combinedConfig as CombinedDockerConfig;
+                    if (dockerConfig != null)
+                    {
+                        var combinedModule = new KubernetesModule<DockerConfig>(m)
+                        {
+                            Config = new DockerConfig(dockerConfig.Image, dockerConfig.CreateOptions)
+                        };
+                        modulesList.Add(combinedModule);
+                        dockerConfig.AuthConfig.ForEach(
+                            auth =>
+                            {
+                                var kubernetesAuth = new ImagePullSecret(auth);
+                                secrets[kubernetesAuth.Name] = kubernetesAuth;
+                            });
+                    }
+                    else
+                    {
+                        throw new InvalidModuleException("Cannot convert combined config into KubernetesModule.");
+                    }
+                }
+            }
+
+            Option<EdgeDeploymentDefinition<DockerConfig>> activeDeployment;
+            try
+            {
+                HttpOperationResponse<object> currentDeployment = await this.client.GetNamespacedCustomObjectWithHttpMessagesAsync(
+                    Constants.K8sCrdGroup,
+                    Constants.K8sApiVersion,
+                    this.deviceNamespace,
+                    Constants.K8sCrdPlural,
+                    resourceName,
+                    cancellationToken: token);
+                string body = JsonConvert.SerializeObject(currentDeployment.Body);
+
+                activeDeployment = currentDeployment.Response.IsSuccessStatusCode ?
+                    Option.Some(this.deploymentSerde.Deserialize(body)) :
+                    Option.None<EdgeDeploymentDefinition<DockerConfig>>();
+            }
+            catch (Exception parseException)
+            {
+                Events.FindActiveDeploymentFailed(resourceName, parseException);
+                activeDeployment = Option.None<EdgeDeploymentDefinition<DockerConfig>>();
+            }
+
+            await this.UpdateImagePullSecrets(secrets, token);
+
+            var metadata = new V1ObjectMeta(name: resourceName, namespaceProperty: this.deviceNamespace);
+            // need resourceVersion for Replace.
+            activeDeployment.ForEach(deployment => metadata.ResourceVersion = deployment.Metadata.ResourceVersion);
+            var customObjectDefinition = new EdgeDeploymentDefinition<DockerConfig>(metaApiVersion, Constants.K8sCrdKind, metadata, modulesList);
+            string customObjectString = this.deploymentSerde.Serialize(customObjectDefinition);
+
+            // the dotnet client is apparently really picky about all names being camelCase,
+            object crdObject = JsonConvert.DeserializeObject(customObjectString);
```

Comment:
Once again, serialize followed by a deserialize...

Reply:
This is converting back from an EdgeDeploymentDefinition to a generic object that is formatted as the client wants.